CMakeLists.txt

```diff
@@ -69,27 +71,48 @@ if(nanopb_BUILD_GENERATOR)
         install(
             FILES ${PROJECT_BINARY_DIR}/${generator_proto_py_file}
                   ${generator_proto_file}
-            DESTINATION ${PYTHON_INSTDIR}/proto/
+            DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/proto/
         )
     endforeach()
 
     install( FILES generator/proto/_utils.py
                    generator/proto/__init__.py
-             DESTINATION ${PYTHON_INSTDIR}/proto/ )
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/proto/ )
+
+    install( FILES generator/nanopb_generator.py
+                   generator/__init__.py
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/ )
+
+    install( FILES generator/__init__.py
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/ )
+
+    # Include the full path to Python executable in Windows .bat scripts, in case it is not in PATH
+    if(WIN32)
+        file(READ generator/protoc-gen-nanopb.bat FILE_CONTENTS)
+        string(REPLACE "python" ${Python_EXECUTABLE} FILE_CONTENTS "${FILE_CONTENTS}")
+        file(WRITE ${PROJECT_BINARY_DIR}/protoc-gen-nanopb.bat "${FILE_CONTENTS}")
+
+        file(READ generator/nanopb_generator.bat FILE_CONTENTS)
+        string(REPLACE "python" ${Python_EXECUTABLE} FILE_CONTENTS "${FILE_CONTENTS}")
+        file(WRITE ${PROJECT_BINARY_DIR}/nanopb_generator.bat "${FILE_CONTENTS}")
+    endif()
 endif()
 
+# Install small script wrappers to invoke the generator from the installed module
 if(WIN32)
     install(
         PROGRAMS
-            generator/nanopb_generator.py
-            generator/protoc-gen-nanopb.bat
+            extra/script_wrappers/nanopb_generator.py
+            ${PROJECT_BINARY_DIR}/protoc-gen-nanopb.bat
+            ${PROJECT_BINARY_DIR}/nanopb_generator.bat
         DESTINATION ${CMAKE_INSTALL_BINDIR}
     )
 else()
     install(
         PROGRAMS
-            generator/nanopb_generator.py
+            extra/script_wrappers/nanopb_generator.py
             generator/protoc-gen-nanopb
+            generator/nanopb_generator
         DESTINATION ${CMAKE_INSTALL_BINDIR}
     )
 endif()
```

Comment:
Should this installs be guarded by the `nanopb_BUILD_GENERATOR`? It looks like the files in `bindir` are useless without the python packages.

Reply:
It should. Fixed.